strawberry/experimental/pydantic/exceptions.py

```diff
@@ -24,3 +25,13 @@ def __init__(self, type: BaseModel):
         )
 
         super().__init__(message)
+
+
+class DefaultAndDefaultFactoryDefined(Exception):
```

Comment:
```suggestion
class BothDefaultAndDefaultFactoryDefinedError(Exception):
```
It is a bit long and ugly, but it is better to be explicit 😊

Also we'll do some work in future to improve errors, so we can always rename them later (especially these ones that are not meant to be caught manually)